[blog] GitLab MR button blog post (#875)
gitpod-io/website: src/routes/blog/gitlab-mr-gitpod-integration.md

---
author: corneliusludmann

> date: Mon Aug 23 2021 08:08:08 GMT+0000 (UTC)

Reviewer: **suggestion**: Since @pawlean aims to merge https://github.com/gitpod-io/website/pull/877 on Thursday, we could publish this later this week or next week for better flow of content. See also [relevant discussion](https://gitpod.slack.com/archives/C01LDA27WJC/p1629721017213400) (internal).

Reviewer 2: Agreed! I would say Monday would be the best time 🚀

excerpt: GitLab has now a Gitpod button on MR pages

> image: teaser-gitlab-cover.png

Reviewer: **suggestion**: Maybe use a header image with a ne kumquat style as in [the previous GitLab related blog post](https://www.gitpod.io/blog/gitlab-integration)? Cc @ChristinFrohne

slug: gitlab-mr-gitpod-integration
subtitle: Launch Gitpod workspace from GitLab MR pages with one click
title: New Gitpod Button has landed on GitLab Merge Request Pages
---

<script context="module">
  export const prerender = true;
</script>

Reviewing merge requests is part of developer's everday life. The quality of code reviews is a crucial aspect of the development process. In reality, it often leads to just skiming the code changes and approving the merge request with a simple “LGTM”. One reason is that doing code reviews needs a context switch for the reviewer. For a carefull review one need to checkout the branch of the merge request, navigate through the code changes, build and test the changed software, … hoping there will be no inferences with ones local developement setup.

This is a situation where ephemeral development environments show their full strength. With Gitpod, you get a fresh development environment just for your review. It has everything set-up for reviewing the changes and will be closed once the job is done.

> ## Launch a Gitpod development environment for a GitLab merge request

Reviewer: **question**: What do you think of using the same phrase here as in the release blog post from GitLab, too? 🦊 

```suggestion
## Launch a preconfigured Gitpod workspace from a GitLab merge request
```

Starting a Gitpod development environment is easy: Just add `gitpod.io/#` in front of a GitLab, GitHub, or Bitbucket URL in your browser's address bar and hit Enter. That's it.

![Gitpod loves GitLab](../../../static/images/blog/gitlab-mr-gitpod-integration/teaser-gitlab-gitpod.jpg)

> With the integrated Gitpod integration in GitLab, it's even easier! Next to the Web IDE button you'll find a Gitpod button in a drop-down menu. With the GitLab 14.2 release you'll find this button even on the merge request pages. Just hit the button and a cloud development environment waits for your review.

Reviewer: **suggestion**: What do you think of mentioning the initial addition of the native integration in GitLab 13.5 released last year in October and also cross-linking to the first blog post related to this integration?
> https://www.gitpod.io/blog/gitlab-integration

> With the integrated Gitpod integration in GitLab, it's even easier! Next to the Web IDE button you'll find a Gitpod button in a drop-down menu. With the GitLab 14.2 release you'll find this button even on the merge request pages. Just hit the button and a cloud development environment waits for your review.

Reviewer: **issue**: Smal typo, right?

```suggestion
With the native Gitpod integration in GitLab, it's even easier! Next to the Web IDE button you'll find a Gitpod button in a drop-down menu. With the GitLab 14.2 release you'll find this button even on the merge request pages. Just hit the button and a cloud development environment waits for your review.
```

> With the integrated Gitpod integration in GitLab, it's even easier! Next to the Web IDE button you'll find a Gitpod button in a drop-down menu. With the GitLab 14.2 release you'll find this button even on the merge request pages. Just hit the button and a cloud development environment waits for your review.

Reviewer: **suggestion**: May I also suggest to include link to the updated Gitpod Integration docs? FYI, the last part [Launch Gitpod in GitLab](https://docs.gitlab.com/ee/integration/gitpod.html#launch-gitpod-in-gitlab) has been updated to reflect the new addition in https://gitlab.com/gitlab-org/gitlab/-/merge_requests/68322. 🤓 
> https://docs.gitlab.com/ee/integration/gitpod.html

![Gitpod button on GitLab merge request](../../../static/images/blog/gitlab-mr-gitpod-integration/create-gitpod-in-mr-view.png)

> ## Contribution to GitLab awarded with “Golden Fork”

Reviewer: **praise**: Woohoo! Well deserved, friend!🎖️

Every release, GitLab selects a community contributor as [most valuable person](https://about.gitlab.com/community/mvp/) (MVP) of the release which is recognized with the prestigious golden fork. In the 14.2 release Cornelius from Gitpod [has been awarded](https://about.gitlab.com/releases/2021/08/22/gitlab-14-2-released/#mvp) for the contribution of the Gitpod integration in GitLab.

![GitLab MVP of release 14.2](../../../static/images/blog/gitlab-mr-gitpod-integration/gitlab-mvp.png)

> ## Further information

Reviewer: **thought**: It could be better to link to that release blog post somewhere in the first two paragraphs of the blog post instead of having a memo down here as it feels disconnected and could go unoticed! Your call. 🤷
